### PR TITLE
FEATURE: RAIL-1129 - Refactoring pivot headers and menus

### PR DESCRIPTION
**Pull request by lhorak**

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1577/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1060/ : FAILURE in 58s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1578/ : SUCCESS in 16s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1061/ : FAILURE in 1m 11s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1584/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1065/ : SUCCESS in 5m 33s

File: src/components/core/pivotTable/ColumnHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+    public onSortRequested = (sortDir: AFM.SortDirection) => {
+        const multiSort = false; // Enable support for multisort with CMD key with 'event.shiftKey';
+        this.props.setSort(sortDir, multiSort);
+    }
+
+    public render() {
+        const { displayName, enableSorting, enableMenu } = this.props;
+        const textAlign = this.getFieldType() === 'a' ? 'left' : 'right';
```
Comment (ocetnik) [inline src/components/core/pivotTable/ColumnHeader.tsx]: 'a' as a constant

File: src/components/core/pivotTable/HeaderCell.tsx [hunk 2]
```diff
@@ -0,0 +1,227 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+import HeaderMenu from './HeaderMenu';
+import { AFM } from '@gooddata/typings';
```
Comment (ocetnik) [inline src/components/core/pivotTable/HeaderCell.tsx]: import @gooddata/typings before HeaderMenu

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1588/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1069/ : SUCCESS in 2m 27s

**Comment by lhorak** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-558.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by lhorak** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1590/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1070/ : SUCCESS in 1m 43s

**Comment by lhorak** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/618/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1591/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/330/ : SUCCESS in 12m 08s

File: src/components/core/pivotTable/ColumnHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
+// (C) 2007-2018 GoodData Corporation
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import { IHeaderParams } from 'ag-grid';
+import { getParsedFields } from '../../../helpers/agGrid';
+import { IHeaderReactComp } from 'ag-grid-react/lib/interfaces';
+import { AFM } from '@gooddata/typings';
```
Comment (ocetnik) [inline src/components/core/pivotTable/ColumnHeader.tsx]: move

File: src/components/core/pivotTable/ColumnGroupHeader.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import * as React from 'react';
```
Comment (ocetnik) [inline src/components/core/pivotTable/ColumnGroupHeader.tsx]: copyright

File: src/components/core/pivotTable/HeaderMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
```
Comment (ocetnik) [inline src/components/core/pivotTable/HeaderMenu.tsx]: copyright

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/350/ : SUCCESS in 14s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1072/ : SUCCESS in 4m 28s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/619/ : SUCCESS in 14m 29s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/264/ : SUCCESS in 13m 53s

**Pull request merged**